src/controller/java/src/chip/devicecontroller/model/ClusterState.java

```diff
@@ -51,6 +57,41 @@ public Optional<Long> getDataVersion() {
     return dataVersion;
   }
 
+  /**
+   * Convenience utility for getting all attributes in Json string format.
+   *
+   * @return all attributes in Json string format., or empty string if not found.
+   */
+  public String getAttributesJson() {
+    JSONObject combinedObject = new JSONObject();
+    ArrayList<JSONObject> attributeJsons =
+        attributes
+            .values()
+            .stream()
+            .map(it -> it.getJson())
+            .filter(it -> it != null)
+            .collect((Collectors.toCollection(ArrayList::new)));
```

Comment:
Do you need to collect it into a separate array list? It seems to me we could iterate over the filtered stream.

Reply:
yes, agree, updated, thanks